tests/integration/pilot/common/routing.go

```diff
@@ -1342,6 +1344,26 @@ spec:
 		viaIngress:       true,
 		workloadAgnostic: true,
 	})
+	t.RunTraffic(TrafficTestCase{
+		name: "optional mTLS",
+		config: gatewayTmpl + httpVirtualServiceTmpl +
+			ingressutil.IngressKubeSecretYAML("cred", "{{.IngressNamespace}}", ingressutil.TLS, ingressutil.IngressCredentialA),
+		templateVars: func(src echo.Callers, dests echo.Instances) map[string]any {
+			params := templateParams(protocol.HTTPS, src, dests, nil)
+			params["GatewayIstioLabel"] = t.Istio.Settings().IngressGatewayIstioLabel
+			params["TLSMode"] = "OPTIONAL_MUTUAL"
+			return params
+		},
+		setupOpts: fqdnHostHeader,
+		opts: echo.CallOptions{
+			Count: 1,
+			Port: echo.Port{
+				Protocol: protocol.HTTPS,
```

Comment:
can we make a call with and without client cert? And maybe can verify it worked by looking at XFCC?

Reply:
I thought about that. But we do not have (can not find any tests that do the tests with certs). Do you have any reference?  I can do follow-up

Reply:
ill look

Reply:
```
							TLS: echo.TLS{
								CertFile:           "/etc/certs/custom/cert-chain.pem",
								KeyFile:            "/etc/certs/custom/key.pem",
								CaCertFile:         "/etc/certs/custom/root-cert.pem",
								InsecureSkipVerify: true,
							},
```

Is one we have. I think its using custom mount, though. But we have them at /cert.crt and /cert.key which should be usable... I guess those may not be accepted,though - just hardcoded self signed ones

Reply:
https://github.com/istio/istio/blob/ff9d5a4134bf806136eb88ca3187e6f6be70741e/tests/integration/security/sds_ingress/util/util.go#L299 looks like what we want

Reply:
ok. thanks . will take a look and follow-up